akka-typed/src/main/scala/akka/typed/receptionist/Receptionist.scala

```diff
@@ -0,0 +1,103 @@
+package akka.typed.receptionist
+
+import akka.typed.ActorRef
+import akka.typed.ActorSystem
+import akka.typed.Extension
+import akka.typed.ExtensionId
+import akka.typed.internal.receptionist.LocalReceptionist
+
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+
+class ReceptionistExt(system: ActorSystem[_]) extends Extension {
+  val receptionist: ActorRef[Receptionist.Command] =
+    ActorRef(
+      system.systemActorOf(
+        LocalReceptionist.behavior // FIXME: either create local or cluster version depending on configuration
+        , "receptionist")(
+        // FIXME: where should that timeout be configured? Shouldn't there be a better `Extension`
+        //        implementation that does this dance for us?
+
+        10.seconds
+      ))
+}
+
+/**
+ * A Receptionist is an entry point into an Actor hierarchy where select Actors
+ * publish their identity together with the protocols that they implement. Other
+ * Actors need only know the Receptionist’s identity in order to be able to use
+ * the services of the registered Actors.
+ */
+object Receptionist extends ExtensionId[ReceptionistExt] {
+  def createExtension(system: ActorSystem[_]): ReceptionistExt = new ReceptionistExt(system)
```

Comment:
`get` is also needed for Java API, otherwise it's wrong type.

`
def get(system: ActorSystem[_]): ReceptionistExt = apply(system)